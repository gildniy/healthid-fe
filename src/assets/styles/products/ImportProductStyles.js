export const ImportProductStyles = {
  paper: {
    width: '70%',
    marginLeft: '15%',
  },
  importContainer: {
    marginLeft: '5%',
    width: '90%',
    marginBottom: '2%'
  },
  textSection: {
    marginTop: '1%',
    marginLeft: '36%',
    fontSize: '20px',
    fontWeight: 'bold',
    color: 'rgba(60, 57, 57, 0.87)'
  },
  typography: {
    fontSize: '1.17em'
  },
  topGrid: {
    marginLeft: '75%'
  },
  childGrid: {
    marginTop: '3%'
  },
  leftFormGroup: {
    marginLeft: '15%',
    width: '50%',
  },
  rightFormGroup: {
    marginLeft: '35%',
    width: '50%',
  },
  templateDivBtn: {
    width: '27%',
    margin: '0 auto',
    marginTop: '2%',
    marginBottom: '2%'
  },
  templateDiv: {
    width: '48%',
    margin: '2% auto',
    color: '#7f7d7d',
    fontSize: '20px'
  },
  templateBtn: {
    backgroundColor: '#424242',
    color: '#FFFFFF',
    padding: '10px 25px 10px 25px',
    width: '120%',
    marginLeft: '-1.5rem',
    borderRadius: '15px',
    marginTop: '20px'
  },
  disabledBtn: {
    borderRadius: '25px',
    backgroundColor: '#A3A3A3',
    color: '#FFFFFF'
  },
  horizontalLine: {
    marginLeft: '-1.5rem',
    border: '0px',
    marginBottom: '0.7em'
  },
};

export const DragFileStyles = {
  text: {
    marginLeft: '26%',
    fontWeight: 'lighter',
    color: '#939393',
    border: '1px solid green'
  },
  fileButton: {
    marginTop: '4%',
    borderRadius: '25px',
    color: '#939393',
    marginLeft: '32%',
    border: '1px solid #939393',
    textTransform: 'inherit',
    width: '35%',
    padding: '2px',
    marginBottom: '5%'
  }
};

export const SubmitFileStyles = {
  submitButton: {
    marginTop: '6%',
    borderRadius: '25px',
    backgroundColor: '#FAF33E',
    color: '#424242',
    fontWeight: 'bold',
    marginLeft: '32%',
    width: '35%',
    padding: '2px',
    marginBottom: '5%'
  }
};

export const SeparatorStyles = {
  mainDiv: {
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
    marginLeft: '13%',
    marginTop: '3%',
    marginBottom: '2%'
  },
  subDiv1: {
    width: '35%'
  },
  horizontalLine: {
    width: '90%'
  },
  subDiv2: {
    width: '23px'
  },
};

export const ProductDuplicationStyles = {
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    padding: '1rem',
  },
  link: {
    textDecoration: 'none',
  },
  optionsLink: {
    width: '25px',
    height: '25px',
    color: 'currentColor',
  },
  backArrowBox: {
    textAlign: 'center',
  },
  backArrow: {
    width: '3rem',
    height: '3rem',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  addButton: {
    paddingTop: '0.5rem'
  },
  addButtonContainer: {
    paddingTop: '0.3rem'
  },
  formTitle: {
    width: '100%',
    padding: '0.5rem 3rem',
    color: '#424242',
    fontSize: '1.3em',
  },
  formUserTitle: {
    padding: '1rem 0 0 2rem',
  },
  tableBox: {
    margin: '0 auto',
  },
  iconBox: {
    display: 'flex',
    alignItems: 'center',
  },
  outletTypeTitle: {
    padding: '1rem',
    backgroundColor: '#A3A3A3',
  },
  outletType: {
    margin: '0 auto',
  },
  addIcon: {
    float: 'right',
  },
  paginate: {
    display: 'flex',
    padding: '3rem',
  },
  previous: {
  },
  spacing: {
    width: '2rem',
  },
  paginateText: {
    fontSize: '1em',
    marginRight: '4rem',
    marginTop: '1rem',
  },
  paginateProductDuplicationText: {
    fontSize: '0.9em',
    marginTop: '0.9rem',
    marginRight: '1.5rem',
    marginLeft: '-1.5rem'
  },
  paginateProduct: {
    display: 'flex',
    marginLeft: '43rem',
  },
  paginateArrow: {
    borderRadius: '0px'
  },
  addLink: {
    float: 'right',
    marginTop: '0px',
  },
  add: {
    marginTop: '0.6rem',
    fontSize: '1em',
  },
};
