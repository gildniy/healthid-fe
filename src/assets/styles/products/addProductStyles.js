export const ProductFormStyles = {
  paperForm: {
    margin: '0 auto 5rem',
    width: '70%',
    height: '100%',
    marginBottom: '50px'
  },
  gridContainer: {
    marginLeft: '5%',
    width: '90%',
    marginBottom: '5%'
  },
  textAreaLabel: {
    color: '#7e7e7e',
    fontSize: '16px'
  },
  textArea: {
    width: '100%',
    border: '1px solid #7e7e7e',
    resize: 'none',
    outline: 'none',
    fontSize: '15px',
    fontWeight: 'lighter',
    padding: '10px'
  },
  childGrid: {
    marginTop: '.2rem',
  },
  descriptionsGrid: {
    marginTop: '2%'
  },
  uploadGrid: {
    marginTop: '2%',
  },
  buttonGrid: {
    marginTop: '4%'
  },
  descriptionField: {
    marginTop: '9%',
  },
  titleStyles: {
    width: '100%',
    position: 'relative'
  },
  title: {
    textAlign: 'center',
    fontSize: '22px',
    fontWeight: 'bold',
    color: '#424242',
    paddingTop: '20px'
  },
  titleLine: {
    width: '110%',
    marginLeft: '-2.4rem',
    backgroundColor: '#E4E4E4',
    height: '2px',
    border: '0px',
  },

  // popper
  rootPaper: {
    marginTop: '2px',
    borderRadius: 0,
    backgroundColor: '#F1F1F1',
    width: '17.4rem'
  },
  widerRootPaper: {
    width: '26.7rem'
  },
  zeroBottomPadding: {
    paddingBottom: 0
  },
  listedCustomers: {
    paddingBottom: 0,
    maxHeight: '18em',
    overflow: 'auto',
  },
  upcContainer: {
    display: 'flex',
    placeContent: 'end space-between',
  },
  upcStyles: {
    width: '48%',
    color: 'red'
  },
};

export const ActionButtonStyles = {
  addButton: {
    borderRadius: '20px',
    border: '1px solid',
    width: '38%',
    margin: 'auto 1%'
  },
  doneButton: {
    borderRadius: '20px',
    width: '58%',
    margin: 'auto 1%'
  },
  cancelButton: {
    borderRadius: '5px',
    border: '1px solid',
    width: '48%',
    marginLeft: '-2rem',
    marginRight: '2.9rem'
  },
  saveButton: {
    borderRadius: '5px',
    width: '48%',
  },
  buttonsGrid: {
    marginLeft: '8.2rem',
    marginRight: '13rem'
  },
  addNewBtn: {
    border: '2px solid #424242',
    color: '#424242',
    borderRadius: '6px',
    padding: '.25rem 2rem',
    fontSize: '12px',
  },
  sendBtn: {
    borderRadius: '6px',
    padding: '.3rem 1rem',
    color: '#fff',
    fontSize: '12px',
    backgroundColor: '#424242',
    marginLeft: '20px',
  },
};

export const TagStyles = {
  container: {
    width: '100%',
    border: '1px solid #ddd',
    padding: '20px 7px',
    borderRadius: '5px',
  }
};

export const ImageUploadStyles = {
  container: {
    width: '150px',
    marginLeft: '45%',
  },
  uploadDiv: {
    height: '150px',
    width: '150px',
    borderRadius: '50%',
    margin: 'auto',
    marginBottom: '10%',
    backgroundColor: '#E4E4E4',
    textAlign: 'center'
  },
  imgPlaceholder: {
    marginTop: '25%',
    marginLeft: '28%',
  },
  uploadedImg: {
    width: '100%',
    height: '100%',
    borderRadius: '5%'
  },
  button: {
    borderRadius: '10%',
    width: '100%',
    color: '#424242',
    fontSize: '100%',
    fontWeight: '200',
    textTransform: 'inherit',
    paddingTop: 0,
    paddingBottom: 0
  },
  label: {
    color: '#A3A3A3',
  },
};

export const BackActionStyles = {
  topDiv: {
    display: 'flex',
    flexDirection: 'row',
    margin: '1rem',
    marginLeft: '10.5rem',
  },
  backIcon: {
    marginTop: '28%',
    color: '#424242'
  },
  header: {
    color: '#424242'
  }
};
