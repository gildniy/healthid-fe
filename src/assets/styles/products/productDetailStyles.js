export const productDetailStyles = theme => ({
  root: {
    width: '100%',
    maxWidth: 'auto',
    backgroundColor: theme.palette.background.paper,
  },
  paper: {
    width: 'auto',
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(6),
    [theme.breakpoints.up(600 + theme.spacing(4))]: {
      width: '65%',
      marginLeft: 'auto',
      marginRight: 'auto',
    }
  },
  card: {
    maxWidth: 345,
  },
  media: {
    height: 180,
    borderRadius: 10
  },
  dividerDiv: {
    marginBottom: '1.5rem',
    marginTop: '1rem',
    backgroundColor: '#E4E4E4',
    height: 'auto'
  },
  dividerHeaders: {
    color: '#424242',
    marginLeft: '50px',
    verticalAlign: 'center'
  },
  newTextFields: {
    marginBottom: '5px'
  },
  descriptionFields: {
    marginBottom: '5px'
  },
  descriptionText: {
    marginTop: '1.1em',
    overflow: 'scroll',
    width: '100%',
    height: '7em',
    wordWrap: 'break-word;'
  },
  productNameStyles: {
    width: '100%',
    position: 'relative'
  },
  productName: {
    textAlign: 'center',
    fontSize: '20px',
    fontWeight: 'bold',
    color: 'rgba(60, 57, 57, 0.87)'
  },
  titleLine: {
    width: '110%',
    marginLeft: '-2.4rem',
    backgroundColor: '#E4E4E4',
    height: '2px',
    border: '0px',
    marginTop: '-0.5rem',
  },
  tagsRoot: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  tagsHeading: {
    marginRight: '80%',
    marginBottom: '4px',
    color: '#898989'
  },
  tagChip: {
    margin: theme.spacing(0.25),
    marginTop: '15px',
    borderRadius: '7px',
    background: '#424242',
    color: 'white',
    marginBottom: '45px'
  },
  batchTextFields: {
    marginBottom: '5px',
  },
  containerGrid: {
    width: '100%',
    marginLeft: '0em',
    padding: '0 2.4em'
  },
  batchDiv: {
    width: '100%',
    marginLeft: '0em',
    padding: '1.2em 2em',
  },
  textContainerGrid: {
    width: '75%',
    marginLeft: '0em',
    padding: '1em'
  },
  childGrids: {
    padding: '1 2em',
    height: '5%'
  },
  arrowButtonGrid: {
    position: 'relative',
    textAlign: 'center',
    marginTop: '1em',
    width: 'auto',
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    marginBottom: theme.spacing(1),
    [theme.breakpoints.up(600 + theme.spacing(4))]: {
      width: '72%',
      marginLeft: 'auto',
      marginRight: 'auto',
    }
  },
  arrowIcon: {
    fontSize: 30,
    color: '#000000',
    cursor: 'pointer'
  },
  buttonsDiv: {
    textAlign: 'right'
  },
  backButton: {
    borderRadius: '7em',
    marginBottom: '50',
    width: '150px',
  },
  buttonMainGrid: {
    width: '100%',
    marginLeft: '0em',
    padding: '1em 1.4em'
  },
  batchRow: {
    height: '25px',
  },
  buttonGrid: {
    position: 'absolute',
    right: '4em',
  },
  approveButton: {
    backgroundColor: '#424242',
    color: 'white',
    borderRadius: '20px',
    padding: '5px 43px'
  },
  tableRow: {
    '&:hover': {
      backgroundColor: '#424242 !import',
    }
  },
  editButton: {
    border: 'thin rgb(66, 66, 66) solid',
    width: '150px',
    borderRadius: '8px',
  },
  approveEditButton: {
    backgroundColor: '#424242',
    color: 'white',
    borderRadius: '10px',
    padding: '5px 43px'
  },
  editProposedButton: {
    marginRight: '20px',
    backgroundColor: '#267EF8',
    color: 'white',
    borderRadius: '10px',
    padding: '5px 50px'
  },
});

export const tableStyles = {
  table: {
    width: '100%',
  },
  batchHeader: {
    boxShadow: '0 0 3px #00000029',
    padding: 0,
    verticalAlign: 'center',
    justifyContent: 'space-between',
    height: 'auto'
  },
  tableHeader: {
    fontSize: '0.8em',
    paddingRight: '0px',
    width: '112px',
    marginRight: '0px',
    color: 'black',
    whiteSpace: 'nowrap'
  },
  tableHeader3: {
    minWidth: '5em',
    maxWidth: '5em',
    padding: 0
  },
  tableHeader1: {
    width: '105px',
    fontSize: '0.8em',
    paddingLeft: '20px',
    paddingRight: '0px',
    marginRight: '0px',
    color: 'black',
    whiteSpace: 'nowrap'
  },
  footerHeader: {
    fontWeight: 500,
    fontSize: '0.87em',
    paddingLeft: '0px',
    paddingBottom: '10px',
    paddingTop: '10px',
    color: 'black'
  },
  tableFooter: {
    width: '70px',
    fontWeight: 900,
    fontSize: '15px',
    color: '#424242',
    border: '1px solid #cccccc',
  },
  batchRow: {
    height: '25px',
  },
  tableCell: {
    width: '112px',
    paddingTop: '12px',
    paddingBottom: '12px',
    paddingRight: '0px',
    marginRight: '0px',
    color: '#424242',
    fontSize: '0.8em',
    whiteSpace: 'nowrap'
  },
  tableCell1: {
    minWidth: '105px',
    paddingLeft: '20px',
    paddingTop: '12px',
    paddingBottom: '12px',
    paddingRight: '0px',
    marginRight: '0px',
    color: '#424242',
    fontSize: '0.8em',
  },
  total: {
    borderStyle: 'none',
    margin: '1.2em 0',
  },
  textGrandTotal: {
    fontWeight: 500,
    fontSize: '0.9em',
    paddingLeft: '2.5rem'
  },
  textGrandTotal1: {
    paddingLeft: '2.7em',
    fontWeight: 500,
    fontSize: '0.9em',
  },
  totalGrid: {
    width: '13.4em',
    paddingTop: '10px',
    paddingBottom: '10px',
    border: '1px solid #EAE8E8',
  },
  textTotal: {
    paddingLeft: '2.5em',
    fontWeight: 500,
    fontSize: '1em',
  },
  textTotal1: {
    fontWeight: 500,
    fontSize: '0.9em',
    paddingLeft: '4.5rem',
  },
  grandTotalGrid: {
    width: '19em',
    paddingTop: '10px',
    marginRight: '0px',
    paddingBottom: '10px',
    border: '1px solid #EAE8E8',
    borderLeft: 'none'
  },
  tableCellDot: {
    paddingTop: '0px',
    paddingBottom: '0px',
    minWidth: '5em',
    maxWidth: '5em',
    paddingLeft: '10px',
    color: '#424242',
  },
  tableCellRight: {
    padding: '5px',
    border: '1px solid #cccccc',
  },
};
