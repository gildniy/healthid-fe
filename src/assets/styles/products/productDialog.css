.table-responsive{
    width: 850px;
}
.table-responsive th{
    height: 60px;
    position: relative;
    color: #424242;
    background: #fff;
    border:none!important;
    
}
.duplicate-form{
    width: 900px;
}

.duplicate-title{
    position: fixed;
    width: 900px;
    margin-left: -2.8rem;
    margin-top: 12.3px;
    padding-bottom: 15px;
    padding-top: 8px;
    z-index: 500;
}
.cell{
    background: #E8E8E8!important;
}
.th-title{
    font-size: 20px;
    font-weight: normal;
}
.th-action{
    position: absolute;
    right: 0;
    top: 0;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 25px;
    justify-content: center;
    align-items: center;
    height: 60px;
    padding-right: 20px;
}

.th-action img{
    width: 20px;
}
.actions{
    cursor: pointer;
}
.actions button, .btns button{
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
}
.t-row{
    padding-left: 30px;
    padding-right: 30px;
    border-bottom: 1px solid #e8e8e8;
    display: grid;
    align-items: center;
    height: 60px;
    position: relative;
}
.Taction{
    grid-template-columns: 1fr 1fr;
}
.btns button{
    height: 40px;
    width: 40px;
    border-radius: 100%;
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
}
.btns button:hover{
    background: #e8e8e8;
    border-radius: 100%;
}

svg.MuiSvgIcon-root-143.paginationIcons {
    width: 0.4em;
}