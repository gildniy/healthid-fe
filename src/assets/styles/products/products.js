export const ProductsStyles = {
  div: {
    backgroundColor: 'white',
    minHeight: '100vh',
    position: 'relative',
  },
};


export const AfterSelectToolBarStyles = {
  iconButton: {
    marginRight: '24px',
    top: '50%',
    display: 'inline-block',
    position: 'relative',
    transform: 'translateY(-50%)',
  },
  icon: {
    color: '#000',
  },
  inverseIcon: {
    transform: 'rotate(90deg)',
  },
};


export const ToolBarStyles = {
  switchBtn: {
    width: '30px',
    height: '30px',
  },
  searchBtn: {
    width: '20px',
    height: '20px',
  },
  eyeButton: {
    width: '20px',
    height: '20px',
  },
  formButton: {
    width: '20px',
    height: '20px',
  },
  plusButton: {
    height: '0.9em',
  },
  iconButton: {
    marginLeft: '25px',
  },
  iconButtonActive: {
    backgroundColor: '#E3E3E3',
    marginLeft: '25px',
  },
  stockButton: {
    height: '0.9em',
  },
  paperForm: {
    padding: '2rem',
    textAlign: 'center',
    marginRight: '1rem',
  },
  paperBtns: {
    marginTop: '3rem',
  },
  openBtn: {
    border: 'thin rgb(66, 66, 66) solid',
    width: '150px',
    marginLeft: '50px',
    color: 'white',
    background: 'rgb(66, 66, 66)',
    borderRadius: '8px',
    '&:hover': {
      color: 'black',
    }
  },
  savePrint: {
    background: 'red',
  },
  closeBtn: {
    border: 'thin rgb(66, 66, 66) solid',
    width: '150px',
    borderRadius: '8px',
  },
  savePrintPaper: {
    width: '30em',
    padding: '1.5em',
    paddingRight: 0
  },
  savePrintPaper1: {
    width: '30em',
    padding: '1.5em',
    paddingTop: '1em',
    paddingRight: 0
  },
  printButton: {
    width: '4em',
    textAlign: 'center'
  },
  savePrintTypo: {
    textAlign: 'center',
    fontSize: '1.2em',
  },
  saveButtonImg: {
    width: '3em',
    height: '2.4em'
  },
  saveButton: {
    padding: '27px',
    paddingTop: '37px',
    marginLeft: '2em'
  },
  manageButton: {
    backgroundColor: '#3A3A3A',
    color: 'white',
    padding: '5px 10px',
    borderRadius: '12px',
    border: 'none',
    marginLeft: '10px',
    cursor: 'pointer',
  },
  menuLink: {
    textDecoration: 'none',
    color: '#424242'
  },
  popper: {
    zIndex: '500',
    marginTop: '15px',
  },
  exportSVG: {
    height: '1em',
  },
  exportSVG2: {
    height: '0.9em',
  },
  switchFormGroup: {
    marginRight: '6rem',
  },
  switchForm: {
    padding: '0rem 2rem',
  },
  searchIcon: {
    width: '20px'
  }
};

export const approveProductsStyles = theme => ({
  root: {
    width: '100%',
    maxWidth: 'auto',
    backgroundColor: theme.palette.background.paper,
  },
  paper: {
    width: 'auto',
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(6),
    [theme.breakpoints.up(600 + theme.spacing(4))]: {
      width: '65%',
      marginLeft: 'auto',
      marginRight: 'auto',
    }
  },
  category: {
    marginTop: '1.1em',
  },
  card: {
    maxWidth: 345,
  },
  media: {
    height: 180,
    borderRadius: 10
  },
  dividerDiv: {
    backgroundColor: '#E4E4E4',
    height: 'auto'
  },
  productNameStyles: {
    width: '100%',
    position: 'relative'
  },
  productName: {
    textAlign: 'center',
    fontSize: '20px',
    fontWeight: 'bold',
    color: 'rgba(60, 57, 57, 0.87)'
  },
  titleLine: {
    width: '110%',
    marginLeft: '-2.4rem',
    backgroundColor: '#E4E4E4',
    height: '2px',
    border: '0px',
    marginTop: '-0.5rem',
  },
  dividerHeaders: {
    marginLeft: '50px',
    verticalAlign: 'center'
  },
  tagChip: {
    margin: theme.spacing(0.25),
    marginTop: '15px',
    borderRadius: '7px',
    background: '#424242',
    color: 'white',
    marginBottom: '45px'
  },
  containerGrid: {
    width: '100%',
    marginLeft: '0em',
    padding: '0em 2.4em'
  },
  arrowButtonGrid: {
    position: 'relative',
    textAlign: 'center',
    marginTop: '1em',
    width: 'auto',
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    marginBottom: theme.spacing(3),
    [theme.breakpoints.up(600 + theme.spacing(4))]: {
      width: '72%',
      marginLeft: 'auto',
      marginRight: 'auto',
    }
  },
  buttonGrid: {
    position: 'absolute',
    right: '4em',
    marginRight: -14
  },
  approveButton: {
    backgroundColor: '#424242',
    color: 'white',
    borderRadius: '10px',
    padding: '5px 43px'
  },
  editButton: {
    marginRight: '20px',
    backgroundColor: '#267EF8',
    color: 'white',
    borderRadius: '10px',
    padding: '5px 50px'
  },
});

export const proposeEditStyles = theme => ({
  arrowButtonGrid: {
    textAlign: 'center',
    marginTop: '1em',
    marginBottom: '-1em',
    width: 'auto',
    [theme.breakpoints.up(600 + theme.spacing(4))]: {
      width: '78%',
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  arrowIcon: {
    fontSize: 30,
    color: '#000000',
    cursor: 'pointer'
  },
  buttonGrid: {
    position: 'absolute',
    right: '15%',
  },
  openBtn: {
    border: 'thin rgb(66, 66, 66) solid',
    paddingLeft: '30px',
    paddingRight: '30px',
    marginLeft: '50px',
    color: 'white',
    background: 'rgb(66, 66, 66)',
    borderRadius: '8px',
    '&:hover': {
      color: 'black',
    }
  },
  closeBtn: {
    border: 'thin rgb(66, 66, 66) solid',
    paddingLeft: '30px',
    paddingRight: '30px',
    borderRadius: '8px',
    background: 'rgba(255, 255, 255, 0)',
  },
});

export default ProductsStyles;
