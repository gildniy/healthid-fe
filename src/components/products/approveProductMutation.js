import gql from 'graphql-tag';

const APPROVE_PRODUCT_MUTATION = gql`
  mutation approveProduct (
    $id: Int!
  ) {
    approveProduct(productId: $id) {
      product {
        id
      }
      success
    }
  }
`;

export default APPROVE_PRODUCT_MUTATION;
