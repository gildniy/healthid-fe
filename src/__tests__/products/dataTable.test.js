import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DataTableWrapper, { DataTable } from '../../components/products/Templates/Table/DataTable';
import * as AppContext from '../../providers/stateProvider';
import { products, columns } from '../../../__fixtures__/products';

const contextValues = { state: { isActive: 'grid3' }, dispatch: jest.fn() };

jest.spyOn(AppContext, 'useStateValue').mockImplementation(() => contextValues);

const props = {
  title: 'Test Products Table',
  data: products,
  onRowClick: jest.fn(),
  loading: false,
  handleSearch: jest.fn(),
  handleViewProposed: jest.fn(),
  currentPath: '',
  handleChangeRowsPerPage: jest.fn(),
  status: 'approved',
  totalCount: jest.fn(),
  handleRequestSort: jest.fn(),
  page: 3,
  columns,
  isSearchActive: true,
  handleChangePage: jest.fn(),
  handleClickSearch: jest.fn(),
  handleHideSearch: jest.fn(),
  session: '',
  rowsPerPage: jest.fn(),
  classes: {},
};

const event = {
  stopPropagation: jest.fn(),
  currentTarget: {
    innerText: 1
  },
  target: {
    value: 'panadol'
  }
};

describe('Products DataTable', () => {
  let wrapper;
 
  describe('when rendering', () => {
    wrapper = shallow(<DataTable {...props} />);
    test('it should render without crushing', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('Check boxes', () => {
    it('when clicking checkboxes', () => {
    wrapper.find('[role="checkbox"]').at(0).simulate('click');
    wrapper.find('[padding="checkbox"]').at(0).childAt(0).simulate('click', event);
    expect(wrapper.find('[padding="checkbox"]').at(0).childAt(0).prop('checked')).toBe(true);
    wrapper.find('[padding="checkbox"]').at(0).childAt(0).simulate('click', event);
    expect(wrapper.find('[padding="checkbox"]').at(0).childAt(0).prop('checked')).toBe(false);
  });

  it('when selecting all check boxes', () => {
    let event = {
      target: { checked: true }
    }
    wrapper.find('WithStyles(TableHeader)').props().onSelectAllClick(event);
    expect(wrapper.find('WithStyles(TableHeader)').prop('numSelected')).toBe(2)
    event = { target: { checked: false } }
    wrapper.find('WithStyles(TableHeader)').props().onSelectAllClick(event);
    expect(wrapper.find('WithStyles(TableHeader)').prop('numSelected')).toBe(0)
  });

  it('handles ClickDeselectAll', () => {
    wrapper.find('TableToolBar').props().handleClickDeselectAll();
    expect(wrapper.find('TableToolBar').prop('numSelected')).toBe(0)
  })
  })

  describe('when searching', () => {
    it('it handles TextChange', () => {
      let event = {target: { value: 'pan' }}
      wrapper.find('TableToolBar').props().handleSearch(event);
      expect(wrapper.find('WithStyles(TableHeader)').prop('rowCount')).toBe(2);
    })
  })

  it('handles RequestSort', () => {
    wrapper.find('WithStyles(TableHeader)').props().onRequestSort('', 'name');
    expect(wrapper.find('WithStyles(TableHeader)').prop('order')).toBe('desc')
    wrapper.find('WithStyles(TableHeader)').props().onRequestSort('', 'named');
    expect(wrapper.find('WithStyles(TableHeader)').prop('orderBy')).toBe('named')
  })
});
