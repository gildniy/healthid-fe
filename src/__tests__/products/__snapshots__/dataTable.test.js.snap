// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products DataTable when rendering it should render without crushing 1`] = `
<div>
  <WithStyles(ForwardRef(Paper))
    elevation={2}
    square={true}
  >
    <TableToolBar
      currentPath=""
      handleClickDeselectAll={[Function]}
      handleClickInverseSelection={[Function]}
      handleEdit={[Function]}
      handleSearch={[MockFunction]}
      handleViewProposed={[MockFunction]}
      name="toolbar"
      numSelected={0}
      title="Test Products Table"
    />
    <div>
      <WithStyles(ForwardRef(Table))
        aria-labelledby="tableTitle"
      >
        <WithStyles(TableHeader)
          headRows={
            Array [
              "id",
              "product name",
              "sku",
            ]
          }
          numSelected={0}
          onRequestSort={[Function]}
          onSelectAllClick={[Function]}
          order="asc"
          orderBy="name"
          rowCount={2}
        />
        <WithStyles(ForwardRef(TableBody))>
          <WithStyles(ForwardRef(TableRow))
            aria-checked={false}
            key="1"
            onClick={[Function]}
            role="checkbox"
            selected={false}
            tabIndex={-1}
          >
            <WithStyles(ForwardRef(TableCell))
              align="center"
              padding="checkbox"
            >
              <WithStyles(CustomCheckbox)
                checked={false}
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": "1.2rem",
                  }
                }
              />
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              <span>
                Panadol
              </span>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              000261
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              <span />
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              <span>
                <Component />
              </span>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              <span>
                <Component />
              </span>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              Stans
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              Stans
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              85
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              408
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              5
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              <Component />
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              <Component />
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              2019-08-13
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
          </WithStyles(ForwardRef(TableRow))>
          <WithStyles(ForwardRef(TableRow))
            aria-checked={false}
            key="2"
            onClick={[Function]}
            role="checkbox"
            selected={false}
            tabIndex={-1}
          >
            <WithStyles(ForwardRef(TableCell))
              align="center"
              padding="checkbox"
            >
              <WithStyles(CustomCheckbox)
                checked={false}
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": "1.2rem",
                  }
                }
              />
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              <span />
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              <span />
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              <span />
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            >
              <span />
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
            <WithStyles(ForwardRef(TableCell))
              align="left"
            />
          </WithStyles(ForwardRef(TableRow))>
        </WithStyles(ForwardRef(TableBody))>
      </WithStyles(ForwardRef(Table))>
    </div>
    <WithStyles(ForwardRef(TablePagination))
      backIconButtonProps={
        Object {
          "aria-label": "Previous Page",
        }
      }
      component="div"
      count={[MockFunction]}
      nextIconButtonProps={
        Object {
          "aria-label": "Next Page",
        }
      }
      onChangePage={[MockFunction]}
      onChangeRowsPerPage={[MockFunction]}
      page={2}
      rowsPerPage={[MockFunction]}
      rowsPerPageOptions={
        Array [
          50,
          100,
          250,
          500,
        ]
      }
    />
  </WithStyles(ForwardRef(Paper))>
</div>
`;
